src/Microsoft.ML.Api/SchemaDefinition.cs

```diff
@@ -319,47 +339,72 @@ public static SchemaDefinition Create(Type userType)
 
             SchemaDefinition cols = new SchemaDefinition();
             HashSet<string> colNames = new HashSet<string>();
-            foreach (var fieldInfo in userType.GetFields())
+
+            var fieldInfos = userType.GetFields();
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.CanWrite && x.GetIndexParameters().Length == 0);
+
+            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+
+            foreach (var memberInfo in memberInfos)
             {
                 // Clause to handle the field that may be used to expose the cursor channel.
                 // This field does not need a column.
                 // REVIEW: maybe validate the channel attribute now, instead
                 // of later at cursor creation.
-                if (fieldInfo.FieldType == typeof(IChannel))
-                    continue;
-                // Const fields do not need to be mapped.
-                if (fieldInfo.IsLiteral)
-                    continue;
+                switch (memberInfo)
+                {
+                    case FieldInfo fieldInfo:
+                        if (fieldInfo.FieldType == typeof(IChannel))
+                            continue;
+
+                        // Const fields do not need to be mapped.
+                        if (fieldInfo.IsLiteral)
+                            continue;
+
+                        break;
 
-                if (fieldInfo.GetCustomAttribute<NoColumnAttribute>() != null)
+                    case PropertyInfo propertyInfo:
+                        if (propertyInfo.PropertyType == typeof(IChannel))
+                            continue;
+                        break;
+
+                    default:
+                        throw Contracts.ExceptNotSupp("Expected a FieldInfo or a PropertyInfo");
+                }
```

Comment:
As far as I understand, this clause is only present because the compiler complains that there must be a default clause, so we actually will never trigger it.
If that's the case, I suggest putting `Contracts.Assert(false)` before throwing, to signify that this is a perceived-never-to-execute code (as opposed to something we don't support not but may in the future).